### ISSUE TITLE
onRequestPermissionResult and onActivityResult support

### ISSUE THREAD

**Issue by bangarharshit** [wealthfront/magellan#63]
`onRequestPermissionResult` and `onActivityResult` support in Screen. These methods propagate the request to activity and then propagate the result back to correct screen. 

It will make it much easier for the implementors of Screen.

**Comment by thepunkprogrammer** [comment]
you can make methods / functions in screen/ basescreenview to pass data to the screen from on activity with when / switch case statements  implemented in onActivityResult or onRequestPermissionResult.  

use language features like abstraction in this case rather getting it implemented in the library.

**Comment by bangarharshit** [comment]
Methods like this should be in the library. Every single app which is non-trivial needs to deal with permission and going to write same useless repetitive logic on their end (it needs a change in both `Activity` and `Screen`).

Also, keeping in library creates a much better abstraction. [Conductor](https://github.com/bluelinelabs/Conductor/blob/develop/conductor/src/main/java/com/bluelinelabs/conductor/Controller.java#L562) does this and is one of the issues as mentioned in this Reddit [thread](https://www.reddit.com/r/androiddev/comments/63z0of/wealthfrontmagellan_the_simplest_navigation/) (I'm not sure why their implementation is so complex, why can't permission or activity result be just delegated to the topmost screen).

The only reason it should not be in the library is that the author doesn't need it and anyone can fork and add it to their version.

**Comment by fabiendevos** [comment]
It's true that we want to be careful about adding too much in the lib, but in this case I agree that this should be part of the library.

**Comment by bangarharshit** [comment]
Agreed. I think a very simple implementation will make life much easier for other devs. I tried implementing it in a PR https://github.com/wealthfront/magellan/pull/64 and would love to make it even more simple.
